js/sagas/session.js

```diff
@@ -5,30 +5,83 @@
  */
 
 import { type Saga } from 'redux-saga'
-import { takeLatest, take, put } from 'redux-saga/effects'
+import { takeLatest, put, select, call } from 'redux-saga/effects'
 import { NavigationActions } from 'react-navigation'
 
-import { IDP_SELECTED, LOGIN_SUCCESS } from '../store/actions/constants'
+import {
+  APPLICATION_INITIALIZED,
+  IDP_SELECTED,
+  LOGIN_SUCCESS,
+  SESSION_INITIALIZE_SUCCESS
+} from '../store/actions/constants'
 import { loadProfile } from '../store/actions/profile'
 import ROUTES from '../navigation/routes'
+import { type GlobalState } from '../reducers/types'
 
-// A saga for the login
-function* loginSaga(): Saga<void> {
-  // Redirect the user to the IdpLoginScreen
-  yield put(
-    NavigationActions.navigate({
-      routeName: ROUTES.AUTHENTICATION_IDP_LOGIN
-    })
-  )
-
-  // Wait for LOGIN_SUCCESS action
-  yield take(LOGIN_SUCCESS)
+function* loginStep(): Saga<void> {
```

Comment:
to make sure nothing is actually returned, I believe you can use the `empty` type ([available but not yet documented](https://github.com/facebook/flow/issues/2225)): `Saga<empty>`

Reply:
Flow doesn't like it.